src/compiler.ts

```diff
@@ -7989,9 +7989,17 @@ export class Compiler extends DiagnosticEmitter {
         );
       }
       case ElementKind.FUNCTION_PROTOTYPE: {
+        let prototypeInstance = <FunctionPrototype>target;
+
+        if (prototypeInstance.is(CommonFlags.STATIC)) {
+          let instance = this.compileFunctionUsingTypeArguments(prototypeInstance, [])!;
```

Comment:
This can return `null`, in which case an `unreachable` should be emitted. Also, providing `alternativeReportNode = propertyAccess` will refer to a better code location in the error case, as it will otherwise report on the original function, which misses a clue where the error actually happened.

Reply:
Should we emit a compiler error here instead of `unreachable()` when the return is `null`? I'm not sure what returning null actually means.

Also the property node fix makes perfect sense. I'll get around to this soon.

Reply:
What happens in situations like these is that some part of the code finds that an operation is invalid, so it emits a diagnostic and returns `null`. That's the case here as well: An error has already been emitted if `compileFunctionUsingTypeArguments` returns `null`.

Reply:
If you are interested in seeing what happens in such a case, make the function being used generic. That'd result in a diagnostic in the resolve step of `compileFunctionUsingTypeArguments`, which returns `null`, making `compileFunctionUsingTypeArguments` return null, leading to an `unreachable` here.

Reply:
That makes perfect sense. Thank you for taking the time to explain this.

Reply:
All set!